test/hotspot/jtreg/compiler/ccp/TestAndConZeroMonotonic.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2025, Google LLC. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8350563
+ * @summary Test we don't prematurely optimize and later unoptimize unprocessed nodes.
+ * @run main/othervm -XX:+UnlockDiagnosticVMOptions -XX:RepeatCompilation=300 -XX:+StressIGVN -XX:+StressCCP -Xcomp -XX:CompileOnly=java.lang.Integer::parseInt compiler.ccp.TestAndConZeroMonotonic
+ * @run main compiler.ccp.TestAndConZeroMonotonic
+ */
+package compiler.ccp;
+
+public class TestAndConZeroMonotonic {
+
+    public static void main(String[] args) {
```

Comment:
Since it's quite an easy test, I suggest to merge the two test files together by calling `Integer::parseInt()` directly instead. You can just add another `@run` statement.

Reply:
Done